### PR TITLE
Fix failing tests for type-reference support

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
This is to support the changes added to runtime APIs for type-reference-support. 
Fixes:
https://github.com/ballerina-platform/ballerina-standard-library/issues/3517

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

**Review by MohamedSabthar — APPROVED** [review]

**Review by niveathika — CHANGES_REQUESTED** [review]
@HindujaB @MohamedSabthar This is a lang update version change, both module version and distribution in Ballerina.toml should reflect this

File: native/src/main/java/io/ballerina/stdlib/serdes/BallerinaStructuredTypeMessageSerializer.java [hunk 1]
```diff
@@ -82,55 +83,62 @@ public Builder generateMessage() {
...
+                messageSerializer.setTableFieldValue((BTable<?, ?>) ballerinaValue);
+                break;
+            case TypeTags.MAP_TAG:
+                @SuppressWarnings("unchecked")
+                BMap<BString, Object> mapValue = (BMap<BString, Object>) ballerinaValue;
+                messageSerializer.setMapFieldValue(mapValue);
+                break;
+            case TypeTags.TYPE_REFERENCED_TYPE_TAG:
+                setFieldValue(ballerinaValue, ((ReferenceType) ballerinaType).getReferredType());
+                break;
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/serdes/BallerinaStructuredTypeMessageSerializer.java]: Is this the correct way to achieve this? Shouldnt the type for refernece type resolved before the setFieldType method itself?

Comment (HindujaB) [reply native/src/main/java/io/ballerina/stdlib/serdes/BallerinaStructuredTypeMessageSerializer.java]: As we are using a `switch-case` to call the suitable `setFieldValue()` method for the type, I think resolving it earlier will also have the same behavior.
And handling it in the switch case might give a small performance advantage for recursive calls.
 
Btw, is it possible to get an `IntersectionType` as a field type here? @MohamedSabthar

Comment (niveathika) [reply native/src/main/java/io/ballerina/stdlib/serdes/BallerinaStructuredTypeMessageSerializer.java]:
```
if type is referencetype
   // recursive get the base type
else
    continue;
   
Switch for set field
```

Checking once if it is a refernce type should be enough. There would be performance hit if this is a refernce type, it is 11 switch cases for recursive reference type. Or else if this is a refernce type, we recursively resolve it and call the setFieldMethod

Comment (HindujaB) [reply native/src/main/java/io/ballerina/stdlib/serdes/BallerinaStructuredTypeMessageSerializer.java]: I'll update the PR

**Review by niveathika — APPROVED** [review]
LGTM

**Comment by MohamedSabthar** [comment]
@HindujaB Found the below getType() and getTupleTypes() API usages

- RecordMessageSerializer - uses getType https://github.com/ballerina-platform/module-ballerina-serdes/blob/main/native/src/main/java/io/ballerina/stdlib/serdes/RecordMessageSerializer.java#L47
- TupleMessageSerializer - uses getTupleTypes https://github.com/ballerina-platform/module-ballerina-serdes/blob/main/native/src/main/java/io/ballerina/stdlib/serdes/TupleMessageSerializer.java#L50

I think we can avoid this [change](https://github.com/ballerina-platform/module-ballerina-serdes/blob/5e350abe1c288c6afeb8f8ab1a82fe6a9c2ff5aa/native/src/main/java/io/ballerina/stdlib/serdes/BallerinaStructuredTypeMessageSerializer.java#L81) by adding the `getReferredType()` in the above places since `entry` (MessageFieldData) always hold a referred type value.

**Pull request merged**